### PR TITLE
Change package name to `Azure.ApplicationModel.Configuration`

### PR DESCRIPTION
**Pull request by maririos**

### CODE REVIEW

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Description>This is the implementation of the Azure SDK for Azure Configuration</Description>
-    <AssemblyTitle>Microsoft Azure Configuration</AssemblyTitle>
+    <Description>This is the implementation of the Azure SDK for Azure App Configuration Service</Description>
+    <AssemblyTitle>Microsoft Azure App Configuration</AssemblyTitle>
```
Comment (KrzysztofCwalina) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: I wonder if we should not standardize here. What about something like the following syntax:
Azure SDK <package_name> component
in this case:
Azure SDK Azure.ApplicationModel.Configuration component

@loarabia, @AlexGhiondea?

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Description>This is the implementation of the Azure SDK for Azure Configuration</Description>
-    <AssemblyTitle>Microsoft Azure Configuration</AssemblyTitle>
+    <Description>This is the implementation of the Azure SDK for Azure App Configuration Service</Description>
```
Comment (KrzysztofCwalina) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: The whole product we create is called Azure SDK. Each dll/package in it is called "Azure SDK Component", and so I would reword it to: Azure SDK Component for Azure App Configuration Service.

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj (hunk 1 above)
Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: I am ok with that. It will show consistency

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: The official name of the service is `Azure App Configuration` we have ApplicationModel. 
Is there a reason why we make that addition?

**Comment by maririos** [comment]
I changed the name of the folder too on preparation for the new repo structure where the folder needs to match the name of the package.
Right now it looks repetitive, but eventually it will change.

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj (hunk 1 above)
Comment (KrzysztofCwalina) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: We will have component for many services under Azure.ApplicationModel. ApplicationModel sounds more like a .NET namespace group, than Azure.App. To me Azure.App sounds like it is the application, not a component for an application. In other words, I tried to make it more idiomatic to how we name namespaces in .NET.

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.Configuration/Azure.Configuration.csproj]: Naming will be more clear when other services are under that namespace.
Aggregating other services under `Azure.ApplicationModel` does sound better than `Azure.Application` . It won't match the service naming 1:1 but close enough.

**Review by AlexGhiondea — APPROVED** [review]
